controllers/reconcile.go

```diff
@@ -515,3 +571,88 @@ func (r *MicroK8sControlPlaneReconciler) scaleDownControlPlane(ctx context.Conte
 	// Requeue so that we handle any additional scaling.
 	return ctrl.Result{Requeue: true}, nil
 }
+
+func createUpgradePod(ctx context.Context, kubeclient *kubernetesClient, nodeName string, nodeVersion string) *corev1.Pod {
+	nodeVersion = strings.TrimPrefix(semver.MajorMinor(nodeVersion), "v")
+	// TODO: Will this image work?
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "upgrade-pod",
+		},
+		Spec: corev1.PodSpec{
+			NodeName: nodeName,
+			Containers: []corev1.Container{
+				{
+					Name:  "upgrade",
+					Image: "sachinsinghcanonical/my-image:latest",
```

Comment:
Is there something special with this image?

Reply:
When I was trying with the curl images, it required a sudo privilege to execute curl to hit the privileged snapd refresh endpoint. So I created this image and it worked fine. I haven't tried the curl images for this.